preload-httpse.js

```diff
@@ -72,3 +74,77 @@ var db = new sqlite3.Database('httpse.sqlite', function (err) {
     }
   })
 })
+
+const rmDir = function (dirPath) {
+  try {
+    var files = fs.readdirSync(dirPath)
+  } catch (e) {
+    return
+  }
+  if (files.length > 0) {
+    for (var i = 0; i < files.length; i++) {
+      var filePath = path.join(dirPath, files[i])
+      if (fs.statSync(filePath).isFile()) {
+        fs.unlinkSync(filePath)
+      } else {
+        rmDir(filePath)
+      }
+    }
+  }
+  fs.rmdirSync(dirPath)
+}
+
+console.log('creating httpse.leveldb')
+rmDir('./httpse.leveldb')
+
+const httpseLevelDB = levelup('httpse.leveldb', {compression: false, errorIfExists: true})
+
+const ruleSets = {}
+for (var id in rulesets.rulesetStrings) {
+  ruleSets[id] = ruleSets[id] || []
+  ruleSets[id].push(rulesets.rulesetStrings[id])
+}
+
+let batch = httpseLevelDB.batch()
+for (var target in rulesets.targets) {
+  let targetRuleSets = []
+  rulesets.targets[target].forEach((id) => {
+    ruleSets[id].forEach((ruleset) => {
```

Comment:
lgtm! eventually the block below should be merged into https://github.com/brave/browser-laptop/blob/3de38bfeec25758f98c556f73c5839a95171c2de/preload-httpse.js#L33-L42 so that the perf benefits apply to the JSON version of the db as well